common/djangoapps/student/helpers.py

```diff
@@ -270,6 +262,31 @@ def get_next_url_for_login_page(request):
     return redirect_to
 
 
+def get_redirect_to(request):
+    """
+    Determine the redirect url and return if safe
+    :argument
+        request: request object
+
+    :returns: redirect url if safe else None
+    """
+    static_file_format = '\w+\.(jpeg|jpg|gif|png|ico)'
```

Comment:
`http.is_safe_url()` will prevent the redirection to any external site or static files e.g. images, using regex we would be able to ensure that the user won't be redirected to any static file on our domain that would be something like as follows:
```
<img src="https://edx.org/login?next=favicon.ico">
```
The reason behind it as per mentioned in the ticket is to make sure that the 3rd party sites can't get information about a user on edx.org. I agree with you that the regex seems pretty weak, as it seems to me as well, but the reason why I added some basic extensions right now was that I wasn't sure which extensions we want to block or allow here. 
So @schenedx @maxrothman @rlucioni it would be great if you guys could help me identify the extensions we want to blacklist or allow here.